code/modules/vehicles/spaceship.dm

```diff
@@ -0,0 +1,234 @@
+
+	var/datum/browser/popup = new(user, "Navigation", "<div align='center'>Navigation</div>")
+	popup.set_content(dat)
+	popup.open()
+
+
+/obj/machinery/computer/navigation/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if("login")
+		if(isAI(usr))
+			authenticated = 2
+			updateUsrDialog()
+			return
+		var/mob/living/carbon/human/C = usr
+		var/obj/item/card/id/I = C.get_active_held_item()
+		if(istype(I))
+			if(check_access(I))
+				authenticated = 1
+			if(ACCESS_MARINE_BRIDGE in I.access)
+				authenticated = 2
+		else
+			I = C.wear_id
+			if(istype(I))
+				if(check_access(I))
+					authenticated = 1
+				if(ACCESS_MARINE_BRIDGE in I.access)
+					authenticated = 2
+	if("logout")
+		authenticated = 0
+
+	if (href_list["UP"])
+		do_orbit_checks("UP")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+
+	else if (href_list["DOWN"])
+		do_orbit_checks("DOWN")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+
+	else if (href_list["escape"])
+		do_orbit_checks("escape")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+		//end the round, xeno minor.
+
+	updateUsrDialog()
+
+
+/obj/machinery/computer/navigation/proc/do_orbit_checks(direction)
+	var/current_orbit = GLOB.orbital_mechanics.current_orbit
+	if(cooldown)
+		return
+
+	if(can_change_orbit(current_orbit, direction))
+		if(direction == "escape")
+			var/message = "The [SSmapping.configs[SHIP_MAP].map_name] is preparing to leave orbit.<br><br>Secure all belongings and prepare for engine ignition."
+			priority_announce(message, title = "Retreat")
+			addtimer(CALLBACK(src, .proc/do_change_orbit, current_orbit, direction), 10 SECONDS)
+
+		var/message = "Prepare for orbital change in 10 seconds.<br><br>Moving [direction] the gravity well.<br><br>Secure all belongings and prepare for engine ignition."
+		priority_announce(message, title = "Orbit Change")
+		addtimer(CALLBACK(src, .proc/do_change_orbit, current_orbit, direction), 10 SECONDS)
+
+/obj/machinery/computer/navigation/proc/can_change_orbit(current_orbit, direction)
+	if(cooldown)
+		to_chat(usr, "The ship is currently recalculating based on previous selection.")
+		return FALSE
+	if(changing_orbit)
+		to_chat(usr, "The ship is currently changing orbit.")
+		return FALSE
+	if(direction == "UP" && current_orbit == ESCAPE_VELOCITY)
+		to_chat(usr, "The ship is already at escape velocity! It is already prepped for the escape jump!")
+		return FALSE
+	if(direction == "DOWN" && current_orbit == SKIM_ATMOSPHERE)
+		to_chat(usr, "WARNING, AUTOMATIC SAFETY ENGAGED. OVERRIDING USER INPUT.")
+		return FALSE
+	if(get_power_amount() <= 500000)
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/navigation/proc/do_change_orbit(current_orbit, direction)
+
+	//chug that sweet sweet powernet juice, like 80% of total
+	if(powered()) //do we still have power?
+		idle_power_usage = 5000
```

Comment:
pretty sure theres a proc called use_power for this

Reply:
for the obj/machinery/power, but this is current obj/machinery/computer
Should this thing be a /power/ computer?